java/test/jmri/jmrix/openlcb/OlcbTestHelper.java

```diff
@@ -0,0 +1,269 @@
+    }
+
+    /** Expects that the next outgoing message (not yet matched with an expectation) is the given
+     * CAN frame.
+     * @param expectedFrame GridConnect-formatted CAN frame.
+     * @param cardinality   how many times to expect this frame, e.g. 'times(2)' or omit for once.
+     */
+    protected void expectFrame(String expectedFrame, VerificationMode cardinality) {
+        class MessageMatchesFrame implements ArgumentMatcher<Message> {
+            private final String frame;
+            private String actual = "";
+
+            public MessageMatchesFrame(String frame) {
+                this.frame = frame;
+            }
+            public boolean matches(Message message) {
+                MessageBuilder d = new MessageBuilder(aliasMap);
+                List<? extends CanFrame> actualFrames = d.processMessage(message);
+                StringBuilder b = new StringBuilder();
+                for (CanFrame f : actualFrames) {
+                    b.append(GridConnect.format(f));
+                }
+                String r = b.toString();
+                if (!frame.equals(r)) {
+                    actual = actual + r;
+                }
+                return frame.equals(r);
+            }
+            public String toString() {
+                //printed in verification errors
+                return "[OpenLCB message with CAN rendering of " + frame + "][actual " + actual + "]";
+            }
+        }
+
+        consumeMessages();
+        verify(outputConnectionMock, cardinality).put(
+                argThat(new MessageMatchesFrame(expectedFrame)),any());
+    }
+
+    protected void expectFrame(String expectedFrame) {
+        expectFrame(expectedFrame, times(1));
+    }
+
+
+    /** Expects that the next outgoing message (not yet matched with an expectation) is the given
+     * message.
+     * @param expectedMessage message that should have been sent to the bus from the local stack.
+     */
+    protected void expectMessage(Message expectedMessage) {
+        consumeMessages();
+        verify(outputConnectionMock).put(eq(expectedMessage), any());
+    }
+
+    protected void expectMessageAndNoMore(Message expectedMessage) {
+        expectMessage(expectedMessage);
+        expectNoMessages();
+    }
+
+    /** Expects that there are no unconsumed outgoing messages. */
+    protected void expectNoFrames() {
+        consumeMessages();
+        verifyNoMoreInteractions(outputConnectionMock);
+        clearInvocations(outputConnectionMock);
+    }
+
+    /** Expects that there are no unconsumed outgoing messages. */
+    protected void expectNoMessages() {
+        expectNoFrames();
+    }
+
+    protected void sendFrameAndExpectResult(String send, String expect) {
+        sendFrame(send);
+        expectFrame(expect);
+        expectNoFrames();
+    }
+
+    protected void sendMessageAndExpectResult(Message send, Message expect) {
+        sendMessage(send);
+        expectMessage(expect);
+    }
+}
```

Comment:
Every file needs to end with an empty line. Once the other issue reported by Static Analysis CI is solved, it will probably complain about this too.

Reply:
fixed